File: src/codes/Users_Manual/link/visualizer_field.cpp
//~#include<autd3.hpp>
//~#include "autd3/link/visualizer.hpp"
//~int main() {
auto autd = autd3::ControllerBuilder({autd3::AUTD3(autd3::Vector3::Zero())})
                .open(autd3::link::Visualizer::builder());

autd3::Vector3 center = autd.geometry().center() + autd3::Vector3(0, 0, 150);
autd3::gain::Focus g(center);
autd.send(g);

autd3::link::PlotConfig config;
config.fname = "xy.png";
autd.link().plot_field(config,
                       autd3::link::PlotRange(center.x() - 20, center.x() + 20,
                                              center.y() - 30, center.y() + 30,
                                              center.z(), center.z(), 1),
                       autd.geometry(), autd3::Segment::S0, 0);
//~return 0; }

File: src/codes/Users_Manual/link/visualizer_field.rs
# use autd3::prelude::*;
# use autd3_link_visualizer::*;
# use std::path::Path;
# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
let mut autd = Controller::builder([AUTD3::new(Vector3::zeros())])
    .open(Visualizer::builder()).await?;

let center = autd.geometry.center() + Vector3::new(0., 0., 150.0 * mm);

autd.send(Focus::new(center)).await?;

autd.link.plot_field(
    PlotConfig {
        fname: Path::new("xy.png").into(),
        ..PlotConfig::default()
    },
    PlotRange{ 
        x_range: center.x - 20.0..center.x + 20.0,
        y_range: center.y - 30.0..center.y + 30.0,
        z_range: center.z..center.z,
        resolution: 1.
    },
    &autd.geometry,
    Segment::S0,
    0
)?;
# autd.close().await?;
# Ok(())
# }
